test/lib/ansible_test/_data/sanity/validate-modules/validate_modules/main.py

```diff
@@ -1453,6 +1453,37 @@ def _validate_argument_spec(self, docs, spec, kwargs, context=None):
                     msg=msg
                 )
 
+            doc_elements = doc_options_arg.get('elements', None)
+            doc_type = doc_options_arg.get('type', 'str')
+            data_elements = data.get('elements', None)
+            if (doc_elements and not doc_type == 'list'):
+                msg = "Argument '%s " % arg
+                if context:
+                    msg += " found in %s" % " -> ".join(context)
+                msg += " defines parameter elements as %s but it is valid only when value of parameter type is list" % doc_elements
+                self.reporter.error(
+                    path=self.object_path,
+                    code='doc-elements-invalid',
+                    msg=msg
+                )
+            if (doc_elements or data_elements) and not (doc_elements == data_elements):
```

Comment:
I'm in favor of #66386 and making elements required, because actually wanting `elements='raw'` seems like the anomaly, but until then the default behavior for data_elements is 'raw', so maybe None and 'raw' should be seen as equivalent?

Reply:
Something like this ?
``((doc_elements or "raw") == (data_elements or "raw"))``

Reply:
I asked mattclay to weigh in on increasing ignore.txt in #66386, and he is also in favor of it so you don't need to change anything about this pull request. Let's merge that one and then this one, so if you could rebase that would be great.